### PR TITLE
options.ip is added

### PR DESCRIPTION
**Pull request by KarthikShanmugasundaram**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @KarthikShanmugasundaram has signed the CLA. Thanks for the pull request!

File: lib/box-client.js [hunk 1]
```diff
@@ -386,7 +386,12 @@ BoxClient.prototype.setIPs = function(ips) {
 	var validIPs = ips.filter(function(ipString) {
 		return isIP(ipString);
 	});
-	this.setCustomHeader(HEADER_XFF, validIPs.join(', '));
+	validIPs = validIPs.join(', ');
```
Comment (KarthikShanmugasundaram) [inline lib/box-client.js]: validIPs are used twice after join(). It is not working for the second statement.

File: lib/box-client.js [hunk 2]
```diff
@@ -386,7 +386,12 @@ BoxClient.prototype.setIPs = function(ips) {
...
+
+	this.setCustomHeader(HEADER_XFF, validIPs);
+
+	this._ipOptions = {};
```
Comment (KarthikShanmugasundaram) [inline lib/box-client.js]: Done.

File: lib/box-client.js [hunk 3]
```diff
@@ -386,7 +386,12 @@ BoxClient.prototype.setIPs = function(ips) {
...
+	this._ipOptions.ip = validIPs;
```
Comment (KarthikShanmugasundaram) [inline lib/box-client.js]: done

File: lib/sessions/anonymous-session.js [hunk 4]
```diff
@@ -50,11 +50,16 @@ function AnonymousSession(config, tokenManager) {
...
  * callback, and then cached for later use.
  *
+ * @param {?TokenRequestOptions} options - Sets optional behavior for the token grant, null for default behavior
  * @param {UpdateTokensCallback} [callback] Called on completion with valid access token or error
  * @returns {void}
  * @private
  */
-AnonymousSession.prototype._refreshAnonymousAccessToken = function(callback) {
+AnonymousSession.prototype._refreshAnonymousAccessToken = function(options, callback) {
+	if (typeof options === 'function') {
```
Comment (KarthikShanmugasundaram) [inline lib/sessions/anonymous-session.js]: done

File: lib/token-manager.js [hunk 5]
```diff
@@ -262,25 +260,37 @@ TokenManager.prototype = {
...
+	 * @param {?TokenRequestOptions} options - Sets optional behavior for the token grant, null for default behavior
 	 * @param {Function} callback - passed a TokenInfo object if tokens were granted successfully
 	 * @returns {void}
 	 */
-	getTokensClientCredentialsGrant: function(callback) {
+	getTokensClientCredentialsGrant: function(options, callback) {
+		if (typeof options === 'function') {
+			callback = options;
+			options = null;
+		}
```
Comment (KarthikShanmugasundaram) [inline lib/token-manager.js]: done

File: lib/box-client.js (hunk 1 above)
Comment (mattwiller) [inline lib/box-client.js]: I mean, you should be able to do something like `var validIPs = ips.filter(ipString => isIP(ipString)).join(', ');` as a one-liner

File: lib/box-node-sdk.js [hunk 6]
```diff
@@ -273,20 +273,22 @@ BoxSDKNode.prototype.getTokensAuthorizationCodeGrant = function(authorizationCod
  * Refreshes the access and refresh tokens for a given refresh token.
  *
  * @param {string} refreshToken - A valid OAuth refresh token
+ * @param {?TokenRequestOptions} options - Sets optional behavior for the token grant, null for default behavior
  * @param {Function} callback - passed a TokenInfo object if tokens were granted successfully
  * @returns {void}
  */
-BoxSDKNode.prototype.getTokensRefreshGrant = function(refreshToken, callback) {
-	this.tokenManager.getTokensRefreshGrant(refreshToken, callback);
+BoxSDKNode.prototype.getTokensRefreshGrant = function(refreshToken, options, callback) {
```
Comment (mattwiller) [inline lib/box-node-sdk.js]: These SDK interfaces are public; I believe they need to be made backwards-compatible via argument shuffling

Comment (KarthikShanmugasundaram) [inline lib/box-node-sdk.js]: done.

File: tests/lib/box-sdk-node-test.js [hunk 7]
```diff
@@ -448,10 +448,21 @@ describe('box-node-sdk', function() {
 
 		it('should call to token manager getTokensRefreshGrant() and propagate result when called', function(done) {
 			var refreshToken = 'rt';
-			sandbox.mock(tokenManagerFake).expects('getTokensRefreshGrant').withExactArgs(refreshToken, done).yields();
+			sandbox.mock(tokenManagerFake).expects('getTokensRefreshGrant').withExactArgs(refreshToken, null, done).yields();
 
-			sdk.getTokensRefreshGrant(refreshToken, done);
+			sdk.getTokensRefreshGrant(refreshToken, null, done);
```
Comment (KarthikShanmugasundaram) [inline tests/lib/box-sdk-node-test.js]: done.

**Pull request merged**